### PR TITLE
applied changes on Crypto module, DRAFT

### PR DESCRIPTION
**Pull request by nandotorterolo**
## Purpose
Replace models with protobuf

## Approach
this PR, only replaces models on **Crypto**, and for this reason, It will not compile.
Next Step: 
replace models in **type classes** module

What I do expect from reviewers is, only focus on Crypto module changes, and evaluate if it  could be potentially the correct implementation of a new  **IRCSNM** Intermediate representation scala crypto for Node model

> We will supporting Ed25519 in tetra.  https://toplprotocol.slack.com/archives/C01TBT00MTJ/p1671196524793219?thread_ts=1671191394.172929&cid=C01TBT00MTJ

## Testing
Unit test in specific package

sbt:typeclasses> project crypto
```
[info] set current project to crypto (in build file:/home/fernando/topl/Bifrost/)
sbt:crypto> compile
Adding existing repo at 'protobuf/src/main/protobuf' to the index
HEAD is now at e48b7bf Bn 582 2313 submodule integration add minting (#11)
[success] Total time: 0 s, completed Dec 19, 2022, 12:01:49 PM
sbt:crypto> test:compile
[warn] sbt 0.13 shell syntax is deprecated; use slash syntax instead: Test / compile
Adding existing repo at 'protobuf/src/main/protobuf' to the index
HEAD is now at e48b7bf Bn 582 2313 submodule integration add minting (#11)
[success] Total time: 0 s, completed Dec 19, 2022, 12:01:53 PM
sbt:crypto> 
[info] Run completed in 5 seconds, 999 milliseconds.
[info] Total number of tests run: 156
[info] Suites: completed 12, aborted 0
[info] Tests: succeeded 156, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 7 s, completed Dec 19, 2022, 12:02:22 PM
sbt:crypto> 
```

## Tickets
*

### CODE REVIEW

File: protobuf/src/main/scala/co/topl/protobuf/utility/Base58.scala [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package co.topl.protobuf.utility
```
Comment (nandotorterolo) [inline protobuf/src/main/scala/co/topl/protobuf/utility/Base58.scala:1]: no changes here. same  file that models package

File: protobuf/src/main/scala/co/topl/protobuf/utility/BinaryTree.scala [hunk 2]
```diff
@@ -0,0 +1,299 @@
+package co.topl.protobuf.utility
```
Comment (nandotorterolo) [inline protobuf/src/main/scala/co/topl/protobuf/utility/BinaryTree.scala:1]: no changes here. same  file that models package

File: protobuf/src/main/scala/co/topl/protobuf/utility/KesBinaryTree.scala [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package co.topl.protobuf.utility
```
Comment (nandotorterolo) [inline protobuf/src/main/scala/co/topl/protobuf/utility/KesBinaryTree.scala:1]: no changes here. same  file that models package

File: protobuf/src/main/scala/co/topl/protobuf/utility/Ratio.scala [hunk 4]
```diff
@@ -0,0 +1,48 @@
+package co.topl.protobuf.utility
```
Comment (nandotorterolo) [inline protobuf/src/main/scala/co/topl/protobuf/utility/Ratio.scala:1]: no changes here. same  file that models package

File: protobuf/src/main/scala/co/topl/protobuf/utility/Sized.scala [hunk 5]
```diff
@@ -0,0 +1,105 @@
+package co.topl.protobuf.utility
```
Comment (nandotorterolo) [inline protobuf/src/main/scala/co/topl/protobuf/utility/Sized.scala:1]: no changes here. same  file that models package

File: protobuf/src/main/scala/co/topl/protobuf/utility/StringDataTypes.scala [hunk 6]
```diff
@@ -0,0 +1,79 @@
+package co.topl.protobuf.utility
```
Comment (nandotorterolo) [inline protobuf/src/main/scala/co/topl/protobuf/utility/StringDataTypes.scala:1]: no changes here. same  file that models package

File: protobuf/src/main/scala/co/topl/protobuf/package.scala [hunk 7]
```diff
@@ -1,4 +1,69 @@
 package co.topl
 
-// TODO BN-714 this package will contain co.topl.models.package object, PR v2
-package object protobuf {}
+import co.topl.proto.models.TypedEvidence
```
Comment (nandotorterolo) [inline protobuf/src/main/scala/co/topl/protobuf/package.scala:3]: TODO, clean up this package object, after the full replacement is done

**Comment by nandotorterolo** [comment]
Sean: SecretKeys are a different set of models from VerificationKeys.  The SecretKeys aren't currently defined as messages in protobuf, so I think those may need to remain in the models module, at least until we add them to protobuf.

**Review by mgrand-topl — APPROVED** [review]
Most of the changes you are making to the crypto module in bifrost will need to be made again to the crypto module in the BramblSc repo.  We need to work together.

Also a change I have made in the BramblSc repo is to rename all of the `Lengths` fields so that they are valid identifiers in other languages. I am prefixing all of the Lengths fields with `bytes`. `32` becomes `bytes32`.

**Comment by nandotorterolo** [comment]
closing, Bifrost Node will consume Bramble Crytpo published models

**Pull request closed without merging**